### PR TITLE
Fixing microwave bug

### PR DESCRIPTION
**Pull request by Lazzi0706**
## About the PR

A bug has been reported in our bug tracker where food can no longer be made from ingredients that were in the microwave at the time the power was lost even if the power comes back on
## Why / Balance

If there the microwave is disconnected from the power supply while cooking food, the contents of the microwave can no longer be used for further cooking because the start button won't work
## Technical details

The problem is due to the fact that items inside a working microwave have an "ActivelyMicrowavedComponent" component, but when there is a power outage it remains. Since the "ActiveMicrowaveComponent" is deleted at powerdown, it was decided to implement the deletion of the ActivelyMicrowavedComponent on the ComponentShutdown side.
## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by deltanedas** [comment]
was there not another pr that did the exact same thing or am i tripping

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 1]
```diff
@@ -108,6 +108,9 @@ private void OnCookStop(Entity<ActiveMicrowaveComponent> ent, ref ComponentShutd
             if (!TryComp<MicrowaveComponent>(ent, out var microwaveComponent))
                 return;
 
+            foreach (var item in _container.GetContainer(ent, "microwave_entity_container").ContainedEntities)
```
Comment (Partmedia) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:111]: Is `_container.GetContainer(ent, "microwave_entity_container")` identical to `microwaveComponent.Storage`?

**Comment by Partmedia** [comment]
> was there not another pr that did the exact same thing or am i tripping

You are referring to: https://github.com/space-wizards/space-station-14/pull/23997

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs (hunk 1 above)
Comment (Lazzi0706) [reply Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs]: > Is `_container.GetContainer(ent, "microwave_entity_container")` identical to `microwaveComponent.Storage`?

Yes, my bad. I redid it

**Comment by Partmedia** [comment]
If you have time, would you mind investigating why the previous fix didn't actually fix the issue, and whether it should be reverted?

**Comment by Partmedia** [comment]
@DrMelon , since you authored https://github.com/space-wizards/space-station-14/pull/23997, would you mind taking a look at this?

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by DrMelon** [comment]
> @DrMelon , since you authored #23997, would you mind taking a look at this?

Absolutely. I'm not sure why the issue seems to persist for power outages; I can't reproduce the bug locally after #23997 myself.
I think the problem highlighted by @EmoGarbage404 is likely the culprit for remaining bugs, as the code there does not pass through StopCooking and so my fix is likely not applying in the case where, for some reason, something external to the microwave removes the active component manually.

**Comment by ShadowCommander** [comment]
Fixed in #23997

**Pull request closed without merging**